source/pervasive/map.rs

```diff
@@ -68,6 +74,13 @@ pub fn axiom_map_insert_different<K, V>(m: Map<K, V>, key1: K, key2: K, value: V
     ensures(equal(m.insert(key2, value).index(key1), m.index(key1)));
 }
 
+#[proof]
+#[verifier(external_body)]
+#[verifier(broadcast_forall)]
+pub fn axiom_map_remove_domain<K, V>(m: Map<K, V>, key: K) {
+    ensures(equal(#[trigger] m.remove(key).dom(), m.dom().remove(key)));
+}
+
```

Comment:
This axiom looks fine.  We should also add something like `axiom_map_insert_different` for `remove`.